### ISSUE TITLE
No Disk Space: IOError: [Errno 2] No usable temporary directory found in ['/tmp', '/var/tmp', '/usr/tmp', '/root']

### ISSUE THREAD

**Issue by cancan101** [matplotlib/matplotlib#5939]
I get the following error when trying to import matplotlib on a machine that has no disk space.:

```
import matplotlib
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 1131, in <module>
    rcParams = rc_params()
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 965, in rc_params
    fname = matplotlib_fname()
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 794, in matplotlib_fname
    configdir = _get_configdir()
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 649, in _get_configdir
    return _get_config_or_cache_dir(_get_xdg_config_dir())
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 626, in _get_config_or_cache_dir
    return _create_tmp_config_dir()
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 550, in _create_tmp_config_dir
    tempdir = tempfile.gettempdir()
  File "/usr/lib/python2.7/tempfile.py", line 269, in gettempdir
    tempdir = _get_default_tempdir()
  File "/usr/lib/python2.7/tempfile.py", line 212, in _get_default_tempdir
    ("No usable temporary directory found in %s" % dirlist))
IOError: [Errno 2] No usable temporary directory found in ['/tmp', '/var/tmp', '/usr/tmp', '/root']
```

I see this on v1.5.1 and 1.4.3.

**Comment by cancan101** [comment]
There is already [a carve out](https://github.com/matplotlib/matplotlib/blob/v1.5.1/lib/matplotlib/__init__.py#L549-L554) for this possibility. I suggest extending this to the `IOError`:
```
    try:
        tempdir = tempfile.gettempdir()
    except NotImplementedError:
        # Some restricted platforms (such as Google App Engine) do not provide
        # gettempdir.
        return None
```

**Comment by cancan101** [comment]
Sorry this was space issue on drive. I will refile correctly.

**Comment by tacaswell** [comment]
That seems reasonable enough to me.

**Comment by KanwarGill** [comment]
@tacaswell Is this supposed to work in the case when there is no space?

**Comment by tacaswell** [comment]
mpl can reasonably run in read-only enviroments (google app engine) so I expect it to be able to run without a useable temp directory due to out-of-space.

It is not a super high priority as if you are out of diskspace you probably have _bigger_ problems than mpl not running, but that is neither here nor there.

**Comment by WeatherGod** [comment]
As I have noted in another issue posting, I have run into situations on
(supposedly locked down) production environments where /home was a separate
partition from where the production work was being done. Another user
inadvertently filled up the tiny /home partition and caused our production
process to grind to a halt because it couldn't create the test file to
check to see if the config directory was writable and errored out. This
logic needs to be revisited at some point.

**Comment by anntzer** [comment]
I am almost certain this is solved by #8939 so closing, but I can't test easily of course...
